Cabal-syntax/src/Distribution/Compat/Prelude.hs

```diff
@@ -243,11 +234,11 @@ import Data.Ord (comparing)
 import Data.Proxy (Proxy (..))
 import Data.Set (Set)
 import Data.String (IsString (..))
+import Data.Typeable (TypeRep, Typeable, typeRep)
 import Data.Void (Void, absurd, vacuous)
 import Data.Word (Word, Word16, Word32, Word64, Word8)
 import Distribution.Compat.Binary (Binary (..))
 import Distribution.Compat.Semigroup (Semigroup (..), gmappend, gmempty)
-import Distribution.Compat.Typeable (TypeRep, Typeable, typeRep)
```

Comment:
Can we remove the module?

Reply:
This was marked as resolved, but the module is still there. Can we not remove it?

Reply:
The build fails without it, FWIW.

Reply:
It builds on my machine and I didn't find any other reference to that module using `git grep`. What error are you getting?

Reply:
I wonder how soon 8.4.4 will fall out of our support window.

Reply:
Looks like the error was introduced by #9900? (And may be specific to old ghcs)

Reply:
Just filed #10226; when this is rebased on that, it should fix the Haddock error.